plugin-server/src/worker/ingestion/event-pipeline/runner.ts

```diff
@@ -215,6 +216,10 @@ export class EventPipelineRunner {
             return this.runHeatmapPipelineSteps(event, kafkaAcks)
         }
 
+        if (event.event === '$ai_generation' || event.event === '$ai_embedding') {
+            event = processAiEvent(event)
+        }
```

Comment:
I believe this should be part of one of the _steps_, maybe `prepareEventStep` (@benjackwhite?)

Reply:
I think this should be its own step and should follow the same step pattern we have (i dont like the pattern but consistency is better than discrepancy). Also it needs to be try/catched _and_ environment var controlled so we can turn it off if it performs poorly.